### PR TITLE
Ensure that series upgrades ignore percona at focal

### PR DESCRIPTION
**Pull request by ajkavanagh**
When doing a series upgrade from bionic to focal, the percona-cluster
charm needs to be avoided.  This filters the applications for percona if
the target is focal.  Note if percona is placed on the same unit as
something else that needs to be upgraded (bad idea) then it may still
get 'upgraded' and the operation will fail.

### CODE REVIEW

File: zaza/openstack/utilities/os_versions.py [hunk 1]
```diff
@@ -278,3 +278,105 @@
...
+        s_newton = CompareOpenstack('newton')
+
+        assert s_newton > s_mitaka
+
+        :returns: <string>
+        """
+        return self._list[self.index]
+
+
+class CompareHostReleases(BasicStringComparator):
```
Comment (AurelienLourot) [inline zaza/openstack/utilities/os_versions.py:373]: What is the difference with `from charmhelpers.core.host.CompareHostReleases`?

File: zaza/openstack/charm_tests/rabbitmq_server/tests.py [hunk 2]
```diff
@@ -26,8 +26,8 @@
 import zaza.openstack.charm_tests.test_utils as test_utils
 import zaza.openstack.utilities.generic as generic_utils
 
-from charmhelpers.core.host import CompareHostReleases
```
Comment (AurelienLourot) [inline zaza/openstack/charm_tests/rabbitmq_server/tests.py:29]: I'm surprised, see my other question, why did we copy this code from charm-helpers in here?

**Review by AurelienLourot — CHANGES_REQUESTED** [review]
LGTM, thanks, just one question about code duplication

File: zaza/openstack/charm_tests/rabbitmq_server/tests.py (hunk 2 above)
Comment (ajkavanagh) [reply zaza/openstack/charm_tests/rabbitmq_server/tests.py:29]: Oh that's deliberate, but perhaps a tad conscientious.  I don't think that zaza-openstack-tests should rely on charmhelpers as a library.  charmhelpers is really meant for running on the unit, and this is a test framework.  CompareHostReleases is a relatively small piece of code and the list of releases (as a concept) is already in os_versions.py in various forms.

Therefore, it actually feels better to drop charmhelpers as a dependency for z-o-t just for this one function (which is useful) and just copy the code in.

Comment (AurelienLourot) [reply zaza/openstack/charm_tests/rabbitmq_server/tests.py:29]: Makes sense, thanks!

**Review by AurelienLourot — APPROVED** [review]

**Pull request merged**